src/addon/addon_manager.cpp

```diff
@@ -402,6 +406,22 @@ AddonManager::install_addon(const AddonId& addon_id)
   }
 }
 
+void
+AddonManager::install_addon_from_local_file(const std::string& filename)
+{
+  const std::string& source_filename = FileSystem::basename(filename);
+  if(!boost::algorithm::ends_with(source_filename, ".zip"))
+    return;
```

Comment:
Maybe it would be better idea to check file magic instead (e.g. see if it starts with 'PK'). I'm not sure what impact it would have on performance, if it makes much difference it's fine how it is already.